app/budget/services/budgetExcelService.js

```diff
@@ -0,0 +1,456 @@
+class BudgetExcelService {
+	constructor (ApiService) { // eslint-disable-line no-unused-vars
+		
+	}
+
+	generateDownload(viewState) {
+
+		// eslint-disable-next-line no-console
+		console.dir(viewState);
+
+		var data = [];
+		var termDescriptions = {
```

Comment:
You should refactor this to instead use the TermService (in sharedApp ) to do these term -> name calculations